src/Events/Custom_Tables/V1/Integrations/ACF/Controller.php

```diff
@@ -0,0 +1,208 @@
+ *
+ * @since   TBD
+ *
+ * @package TEC\Events\Custom_Tables\V1\Integrations\ACF;
+ */
+
+namespace TEC\Events\Custom_Tables\V1\Integrations\ACF;
+
+use tad_DI52_ServiceProvider as Service_Provider;
+
+/**
+ * Class Controller.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Events\Custom_Tables\V1\Integrations\ACF;
+ */
+class Controller extends Service_Provider {
+	/**
+	 * The priority at which the ACF field handling will start.
+	 *
+	 * @since TBD
+	 */
+	public const EARLY_PRIORITY = 0;
+
+	/**
+	 * The priority at which the ACF field handling will end.
+	 *
+	 * @since TBD
+	 */
+	public const LATE_PRIORITY = 1000;
+
+	/**
+	 * The priority at which the ACF field handling will start for AJAX queries.
+	 *
+	 * @since TBD
+	 */
+	public const AJAX_QUERY_PRIORITY = 10;
+
+	/**
+	 * A flag property indicating whether an ACF field of the supported type is being rendered or not.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	private bool $rendering_acf_field = false;
+
+	/**
+	 * Returns the list of supported field types.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string> The list of supported field types.
+	 */
+	public static function get_supported_field_types(): array {
+		return [
+			'post_object',
+			'relationship',
+		];
+	}
+
+	/**
+	 * Registers the implementations, actions and filters required by the Custom Tables implementation to work with
+	 * the Advanced Custom Fields plugin.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register() {
+		$this->container->singleton( self::class, $this );
+
+		foreach ( self::get_supported_field_types() as $supported_field_type ) {
+			// Detect the start of a specific ACF field type to boot the handling.
+			add_action( 'acf/render_field/type=' . $supported_field_type, [
+				$this,
+				'start_acf_handling'
+			], self::EARLY_PRIORITY, 0 );
+			// Hook again reasonably late to stop the handling.
+			add_action( 'acf/render_field/type=' . $supported_field_type, [
+				$this,
+				'stop_acf_handling'
+			], self::LATE_PRIORITY, 0 );
+		}
+
+		// Hook to redirect queries in the context of AJAX queries. The below filter is used only in AJAX context.
+		add_filter( 'acf/fields/post_object/query', [
+			$this,
+			'redirect_post_ajax_query'
+		], self::AJAX_QUERY_PRIORITY, 2 );
+
+		// Register a dedicated query modified implementation, pull the flag from the provider.
+		$this->container->bind(
+			Query_Modifier::class, fn() => ( new Query_Modifier() )->set_handling( $this->rendering_acf_field )
+		);
+		add_filter( 'tec_events_custom_tables_v1_query_modifier_implementations', [
+			$this,
+			'add_query_modifier_implementation'
+		] );
+	}
+
+	/**
+	 * Removes the actions and filters added by the provider.
+	 *
+	 * @since TBD
+	 *
+	 * @return void Actions and filters are removed.
+	 */
+	public function unregister() {
+		foreach ( self::get_supported_field_types() as $supported_field_type ) {
+			// Detect the start of a specific ACF field type to boot the handling.
```

Comment:
This inline comment and the one below make sense if you know the context, but less so if you just hit them - as they read like you're adding the actions, not removing them.

Just a thought, not a needed fix.